internal/internal_event_handlers.go

```diff
@@ -881,6 +886,41 @@ func (wc *workflowEnvironmentImpl) SideEffect(f func() (*commonpb.Payloads, erro
 	wc.logger.Debug("SideEffect Marker added", tagSideEffectID, sideEffectID)
 }
 
+func (wc *workflowEnvironmentImpl) TryUse(flag sdkFlag) bool {
+	return wc.sdkFlags.tryUse(flag, !wc.isReplay)
+}
+
+func (wc *workflowEnvironmentImpl) QueueUpdate(name string, f func()) {
+	wc.blockedUpdates[name] = append(wc.blockedUpdates[name], f)
+}
+
+func (wc *workflowEnvironmentImpl) HandleUpdates(name string) bool {
+	updatesHandled := false
+	if blockedUpdates, ok := wc.blockedUpdates[name]; wc.sdkFlags.tryUse(SDKPriorityUpdateHandling, !wc.isReplay) && ok {
+		updatesHandled = true
+		for _, update := range blockedUpdates {
+			update()
+		}
+		delete(wc.blockedUpdates, name)
+	}
+	return updatesHandled
+}
+
+func (wc *workflowEnvironmentImpl) DrainUnhandledUpdates() bool {
+	rerun := false
+	// Check if any blocked updates remain when we have no more coroutines to run and let them run so they are rejected.
+	// Generally iterating a map in workflow code is bad because it is non deterministic
+	// this case is fine since all these update handles will be rejected and not recorded in history.
+	for name, us := range wc.blockedUpdates {
+		for _, u := range us {
+			u()
+		}
+		rerun = true
```

Comment:
Is it definitely true that `us` was non-empty?

Reply:
Yes, but i'll admit that isn't  obvious. We could move the  line into the range block to cover that case.